### PR TITLE
Slack v2 mirroring

### PR DESCRIPTION
**Pull request by abaumgarten**
## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/29425

## Description
Fixed an issue where exception in invite to mirrored channel function was not caught.

## Minimum version of Demisto
- [x] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/Slack/.secrets-ignore [hunk 1]
```diff
@@ -16,3 +16,8 @@ https://api.slack.com
 support@demisto.com
 cool_user@coolorg.com
 https://raw.githubusercontent.com
+re.findall(USER_TAG_EXPRESSION,
```
Comment (liorblob) [inline Packs/Slack/.secrets-ignore:19]: remove those, run `secrets` with `-ie` flag

File: Packs/Slack/Integrations/Slack/Slack.py [hunk 2]
```diff
@@ -773,8 +771,11 @@ def check_for_mirrors():
                     users: List[Dict] = demisto.mirrorInvestigation(investigation_id,
                                                                     f'{mirror_type}:{direction}', auto_close)
                     if mirror_type != 'none':
-                        invited_users = invite_to_mirrored_channel(channel_id, users)
-                        updated_users.extend(invited_users)
+                        try:
+                            invited_users = invite_to_mirrored_channel(channel_id, users)
+                            updated_users.extend(invited_users)
+                        except Exception as error:
+                            demisto.error(f"There was an error while invite to mirror channel:  {error}")
```
Comment (liorblob) [inline Packs/Slack/Integrations/Slack/Slack.py:778]:
```suggestion
                            demisto.error(f"Could not invite investigation users to the mirrored channel:  {error}")
```

File: Packs/Slack/Integrations/Slack/Slack_test.py [hunk 3]
```diff
@@ -311,6 +310,57 @@
...
+                                                                       'username': 'spengler'},
+                                                                      {'email': 'perikles@acropoli.com',
+                                                                       'username': 'perikles'}])
+    mocker.patch.object(Slack, 'invite_to_mirrored_channel', side_effect=Exception)
+    with capfd.disabled():
+        try:
+            check_for_mirrors()
+        except Exception:
+            assert False
+
```
Comment (liorblob) [inline Packs/Slack/Integrations/Slack/Slack_test.py:362]: Assert that `setIntegrationContext ` has been called

File: Packs/Slack/ReleaseNotes/1_3_7.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### Slack v2
+- Fixed an issue where exception in invite to mirrored channel function was not caught.
+- Upgraded the Docker image to demisto/slack:1.0.0.12410.
```
Comment (liorblob) [inline Packs/Slack/ReleaseNotes/1_3_7.md:5]: No need for the docker note. Either replace with `Upgraded slack API version` or `Stability improvements`

**Review by liorblob — CHANGES_REQUESTED** [review]
Nice, see comments :)

File: Packs/Slack/ReleaseNotes/1_3_7.md (hunk 4 above)
Comment (guykeller) [reply Packs/Slack/ReleaseNotes/1_3_7.md:5]: This is a new content convention, docker image upgrades must be explicitly mentioned

Comment (abaumgarten) [reply Packs/Slack/ReleaseNotes/1_3_7.md:5]: Done X2

File: Packs/Slack/Integrations/Slack/Slack_test.py [hunk 5]
```diff
@@ -311,6 +310,57 @@
 '''
 
 
+def test_exception_in_invite_to_mirrored_channel(mocker, capfd):
```
Comment (Itay4) [inline Packs/Slack/Integrations/Slack/Slack_test.py:313]: avoid using `capfd` as we might miss stdout or stderr outputs, you should mock the relevant log functions instead

File: Packs/Slack/Integrations/Slack/Slack_test.py (hunk 3 above)
Comment (liorblob) [reply Packs/Slack/Integrations/Slack/Slack_test.py:362]: there should not be a try/catch here, as no error is being thrown.

**Review by liorblob — APPROVED** [review]

**Pull request merged**